server/src/main/java/thesistrack/ls1/service/MailingService.java

```diff
@@ -10,217 +10,242 @@
+import thesistrack.ls1.exception.MailingException;
 
-import java.io.File;
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.text.SimpleDateFormat;
-import java.util.Arrays;
+import java.time.Instant;
+import java.util.*;
 
 @Service
 public class MailingService {
-
     private final JavaMailSender javaMailSender;
-    private final FileSystemStorageService storageService;
-    private final String environment;
+    private final UploadService uploadService;
+
+    private final boolean enabled;
+    private final String mailFooter;
+    private final String workspaceUrl;
     private final String sender;
-    private final String chairMemberRecipientsList;
-    private final Path rootLocation;
+    private final List<String> chairMemberRecipientsList;
+    private final Path mailTemplateLocation;
 
     @Autowired
-    public MailingService(final JavaMailSender javaMailSender,
-                          final FileSystemStorageService storageService,
-                          @Value("${thesis-track.environment}") String environment,
-                          @Value("${thesis-track.mail.sender}") String sender,
-                          @Value("${thesis-track.mail.chair-member-recipients}") String chairMemberRecipientsList,
-                          @Value("${thesis-track.storage.mailing-templates-location}") String mailingTemplatesLocation) {
+    public MailingService(
+            JavaMailSender javaMailSender,
+            UploadService uploadService,
+            @Value("${thesis-track.mail.enabled}") boolean enabled,
+            @Value("${thesis-track.mail.mail-template-location}") String mailTemplateLocation,
+            @Value("${thesis-track.mail.sender}") String sender,
+            @Value("${thesis-track.mail.chair-member-recipients}") String chairMemberRecipientsList,
+            @Value("${thesis-track.mail.footer}") String mailFooter,
+            @Value("${thesis-track.mail.workspace-url}") String workspaceUrl
+    ) {
         this.javaMailSender = javaMailSender;
-        this.storageService = storageService;
-        this.environment = environment;
+        this.uploadService = uploadService;
+
+        this.enabled = enabled;
         this.sender = sender;
-        this.chairMemberRecipientsList = chairMemberRecipientsList;
-        this.rootLocation = Paths.get(mailingTemplatesLocation);
-    }
+        this.workspaceUrl = workspaceUrl;
+        this.mailFooter = mailFooter;
+        this.mailTemplateLocation = Paths.get(mailTemplateLocation);
 
-    public String getMailTemplate(final String filename) {
-        return storageService.readFromFile(rootLocation, filename + ".html");
+        if (chairMemberRecipientsList != null && !chairMemberRecipientsList.isEmpty()) {
+            this.chairMemberRecipientsList = Arrays.asList(chairMemberRecipientsList.split(";"));
```

Comment:
Couldn't you use your `StringToArrayConverter` here? ;)

Reply:
Mmh it has a different separator and still exist because of legacy code. The StringToArrayConverter is mainly for request parameters and not configuration but I will consider refactoring this later